python_telnet_vlc/vlctelnet.py

```diff
@@ -36,27 +36,65 @@ class ConnectionError(Exception):
     pass
 
 
+class AuthError(Exception):
+    """Failed to login with provided password."""
+    pass
+
+
 # VLC Telnet Class
 class VLCTelnet(object):
     """Conection to VLC using Telnet."""
 
     # Non commands
-    def __init__(self, host="localhost", password="admin", port=4212):
-        # Connect to telnet. Host and port are __init__ arguments
+    def __init__(self, host="localhost", password="admin", port=4212, connect=True, login=True):
+        self.tn = telnetlib.Telnet()
+        self.host = host
+        self.port = port
+        self.password = password
+        if connect:
+            self.connect()
+        # Login to VLC using password provided in the arguments
+        if connect and login:
+            self.login()
+
+    def connect(self):
+        """Connect to VLC."""
+        # Connect to telnet.
         try:
-            self.tn = telnetlib.Telnet(host, port=port)
+            self.tn.open(self.host, port=self.port, timeout=10)
         except sockerr:
             raise ConnectionError("Could not connect to VLC. Make sure the Telnet interface is enabled and accessible.")
-        # Login to VLC using password provided in the arguments
-        self.tn.read_until(b"Password: ")
-        self.run_command(password, False)
 
-    def run_command(self, command, log=True):
+    def disconnect(self):
+        """Disconnect from VLC."""
+        self.tn.close()
+        # reset the instance
+        self.tn = telnetlib.Telnet()
+
+    def login(self):
+        """Login with password."""
+        self.tn.read_until(b"Password: ")
+        full_command = self.password.encode('utf-8') + b'\n'
+        self.tn.write(full_command)
+        while True:
```

Comment:
Nit: Should we replace with a finite loop and throw an error if the procedure isn't finished within a certain number (10?) iterations to avoid a possible infinite loop?

Reply:
Sounds good!